domain/src/test/java/org/oppia/domain/onboarding/AppStartupStateControllerTest.kt

```diff
@@ -42,16 +49,17 @@ import org.oppia.util.logging.GlobalLogLevel
 import org.oppia.util.logging.LogLevel
 import org.oppia.util.threading.BackgroundDispatcher
 import org.oppia.util.threading.BlockingDispatcher
+import org.robolectric.Shadows.shadowOf
 import org.robolectric.annotation.Config
 import javax.inject.Inject
 import javax.inject.Qualifier
 import javax.inject.Singleton
 import kotlin.coroutines.EmptyCoroutineContext
 
-/** Tests for [OnboardingFlowController]. */
+/** Tests for [AppStartupStateController]. */
 @RunWith(AndroidJUnit4::class)
 @Config(manifest = Config.NONE)
-class OnboardingFlowControllerTest {
+class AppStartupStateControllerTest {
```

Comment:
Tests to add:
- Initial app state, deprecation enabled, not deprecated before deprecation date
- Initial app state, deprecation enabled, deprecated after deprecation date
- Initial app state, deprecation disabled, not deprecated before deprecation date
- App opened before, onboarding flow incomplete, deprecation enabled, not deprecated before deprecation date
- App opened before, onboarding flow incomplete, deprecated after deprecation date
- App opened before, onboarding flow completed, deprecation enabled, not deprecated before deprecation date
- App opened before, onboarding flow completed, deprecated after deprecation date